configure.ac

```diff
@@ -722,6 +722,10 @@ fi
 ], [AC_MSG_ERROR([Unknown value for --with-crypto])]
 )
 
+if test "$with_crypto" = "openssl" || test "$with_crypto" = "internal" || test "$with_crypto" = "" ; then
+	AC_DEFINE([HAVE_OSPF6_AT], [1], [Enable ospf6 authentication trailer])
+fi
+
```

Comment:
I don't really understand this test on two points ...

First, "internal" should always be there, right? So HAVE_OSPF6_AT will defined in all compilations of FRR?

Second, the third test seems to say "if there is no crypto engine available, go ahead and define HAVE_OSPF6_AT" ... which doesn't make sense to me.

==
Overall, what is the goal of HAVE_OSPF6_AT as a define? Why separate the code out throughout using ifdef conditionally compiling based on this define? Maybe if we understood the goal better here, then we might be able to figure out a better way to do this?

Reply:
I was trying to control the feature with compile time macro, Later found that all hash algos are not supported as part of internal_crypto macro, So in future if internal_crypto also becomes options, wanted to handle all these scenarios and kept the macro.
What would be your suggestions ?

Reply:
I don't think we need to have the `ifdef` scattered throughout ... the macro might make sense to set a variable to determine what kinds of crypto are available, but removing or adding code at compile isn't something we've done this much of before ...

Reply:
I have reverted the changes for removing feature compile macro as per suggestion and updated the changes. Can you continue reviewing the new code uploaded.